### PR TITLE
Added a new method SetUserMetadata to the PostPolicy

### PR DESCRIPTION
**Pull request by wowi42**
Hi there,

here is a PR for a new feature I need for a project I am working on with @hnb2.

# Feature

```SetUserSpecifiedMetadata(key string, value string)``` allows you to add user specified metadata following this guide which mention:

| key | explanation|
| ---  | ----------- |
| x-amz-meta-* | User-specified metadata.This condition supports exact matching and starts-with condition match type. |

### CODE REVIEW

File: Minio/DataModel/PostPolicy.cs [hunk 1]
```diff
@@ -158,6 +158,22 @@ public void SetSuccessStatusAction(string status)
...
+        public void SetUserSpecifiedMetadata(string key, string value)
+        {
+            if (string.IsNullOrEmpty(key))
+                throw new ArgumentException("Key is Empty");
+            if (string.IsNullOrEmpty(value))
+                throw new ArgumentException("Value is Empty");
+            string headerName = "x-amz-meta-" + key;
+            this.policies.Add(new Tuple<string, string, string>("eq", "$" + headerName, value));
+            this.formData.Add(headerName, value);
+        }
```
Comment (poornas) [inline Minio/DataModel/PostPolicy.cs]: The SetKey() method already allows the user to set form data, and the custom metadata can be passed in with the x-amz-meta prefix. Is there a reason why an additional method would help

Comment (harshavardhana) [reply Minio/DataModel/PostPolicy.cs]: It is coming from here https://github.com/minio/minio-go/pull/839

**Comment by harshavardhana** [comment]
> SetUserSpecifiedMetadata(key string, value string) allows you to add user specified metadata following this guide which mention:

SetUserMetadata(key string, value string) is a better API IMO.  Borrowing suggestion by @vadmeste on minio-go PR.

**Comment by wowi42** [comment]
Done

**Pull request merged**